openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/DataContractRepository.java

```diff
@@ -0,0 +1,197 @@
+/*
+ *  Copyright 2021 Collate
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.openmetadata.service.jdbi3;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.openmetadata.schema.entity.data.DataContract;
+import org.openmetadata.schema.type.EntityReference;
+import org.openmetadata.schema.type.Include;
+import org.openmetadata.schema.type.Relationship;
+import org.openmetadata.service.Entity;
+import org.openmetadata.service.resources.data.DataContractResource;
+import org.openmetadata.service.util.EntityUtil;
+import org.openmetadata.service.util.EntityUtil.Fields;
+
+@Repository
+public class DataContractRepository extends EntityRepository<DataContract> {
+
+  private static final String DATA_CONTRACT_UPDATE_FIELDS =
+      "entity,owners,status,schema,qualityExpectations,contractUpdates";
+  private static final String DATA_CONTRACT_PATCH_FIELDS =
+      "entity,owners,status,schema,qualityExpectations,contractUpdates";
+
+  public DataContractRepository() {
+    super(
+        DataContractResource.COLLECTION_PATH,
+        Entity.DATA_CONTRACT,
+        DataContract.class,
+        Entity.getCollectionDAO().dataContractDAO(),
+        DATA_CONTRACT_PATCH_FIELDS,
+        DATA_CONTRACT_UPDATE_FIELDS);
+  }
+
+  @Override
+  public void setFullyQualifiedName(DataContract dataContract) {
+    String entityFQN = dataContract.getEntity().getFullyQualifiedName();
+    String name = dataContract.getName();
+    dataContract.setFullyQualifiedName(entityFQN + ".dataContract_" + name);
+  }
+
+  @Override
+  public void setFields(DataContract dataContract, Fields fields) {}
+
+  @Override
+  public void clearFields(DataContract dataContract, Fields fields) {}
+
+  @Override
+  public void prepare(DataContract dataContract, boolean update) {
+    EntityReference entityRef = dataContract.getEntity();
+
+    if (!update) {
+      validateEntityLink(entityRef);
+    }
+    validateSchemaFieldsAgainstEntity(dataContract, entityRef);
```

Comment:
should this validation happen at creation time? Or would it make sense for someone to create a data contract that today is not passing, but that we can work on getting fixed in the schema, etc.?

We can leave the contract status as "failed" but not sure i'd raise the `IllegalArgumentException`